haystack/components/rankers/sentence_transformers_diversity.py

```diff
@@ -0,0 +1,237 @@
+    ranker.warm_up()
+
+    docs = [Document(content="Paris"), Document(content="Berlin")]
+    query = "What is the capital of germany?"
+    output = ranker.run(query=query, documents=docs)
+    docs = output["documents"]
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str = "sentence-transformers/all-MiniLM-L6-v2",
+        top_k: int = 10,
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),
+        similarity: Literal["dot_product", "cosine"] = "cosine",
+        query_prefix: str = "",
+        query_suffix: str = "",
+        document_prefix: str = "",
+        document_suffix: str = "",
+        meta_fields_to_embed: Optional[List[str]] = None,
+        embedding_separator: str = "\n",
+    ):
+        """
+        Initialize a SentenceTransformersDiversityRanker.
+
+        :param model: Local path or name of the model in Hugging Face's model hub,
+            such as `'sentence-transformers/all-MiniLM-L6-v2'`.
+        :param top_k: The maximum number of Documents to return per query.
+        :param device: The device on which the model is loaded. If `None`, the default device is automatically
+            selected.
+        :param token: The API token used to download private models from Hugging Face.
+        :param similarity: Similarity metric for comparing embeddings. Can be set to "dot_product" (default) or
+            "cosine".
+        :param query_prefix: A string to add to the beginning of the query text before ranking.
+            Can be used to prepend the text with an instruction, as required by some embedding models,
+            such as E5 and BGE.
+        :param query_suffix: A string to add to the end of the query text before ranking.
+        :param document_prefix: A string to add to the beginning of each Document text before ranking.
+            Can be used to prepend the text with an instruction, as required by some embedding models,
+            such as E5 and BGE.
+        :param document_suffix: A string to add to the end of each Document text before ranking.
+        :param meta_fields_to_embed: List of meta fields that should be embedded along with the Document content.
+        :param embedding_separator: Separator used to concatenate the meta fields to the Document content.
+        """
+        torch_and_sentence_transformers_import.check()
+
+        self.model_name_or_path = model
+        if top_k is None or top_k <= 0:
+            raise ValueError(f"top_k must be > 0, but got {top_k}")
+        self.top_k = top_k
+        self.device = ComponentDevice.resolve_device(device)
+        self.token = token
+        self.model = None
+        if similarity not in ["dot_product", "cosine"]:
+            raise ValueError(f"Similarity must be one of 'dot_product' or 'cosine', but got {similarity}.")
```

Comment:
Could we add a test checking for this ValueError

Reply:
We already test this in [`test_run_incorrect_similarity()`](https://github.com/deepset-ai/haystack/blob/0d923b211ab4535fb4711ec716ed59bf8b3c7e1b/test/components/rankers/test_sentence_transformers_diversity.py#L130). 

I will update it so that it also matches for `"Similarity must be one of 'dot_product' or 'cosine', but got {similarity}."`

Reply:
Ahh okay sorry for the false positive then. I was checking code line coverage using the [coverage report](https://github.com/deepset-ai/haystack/pull/7095#issuecomment-1964038833) and I had thought it said this line wasn't covered for some reason.